src/o365/spo/commands/listitem/listitem-record-declare.ts

```diff
@@ -0,0 +1,309 @@
+import auth from "../../SpoAuth";
+import config from "../../../../config";
+import commands from "../../commands";
+import GlobalOptions from "../../../../GlobalOptions";
+import * as request from "request-promise-native";
+import {
+  CommandOption,
+  CommandValidate,
+  CommandTypes,
+  CommandError
+} from "../../../../Command";
+import SpoCommand from "../../SpoCommand";
+import Utils from "../../../../Utils";
+import { Auth } from "../../../../Auth";
+import {
+  ContextInfo,
+  ClientSvcResponse,
+  ClientSvcResponseContents,
+} from "../../spo";
+import { ClientSvc, IdentityResponse } from "../../common/ClientSvc";
+
+const vorpal: Vorpal = require("../../../../vorpal-init");
+
+interface CommandArgs {
+  options: Options;
+}
+
+interface Options extends GlobalOptions {
+  webUrl: string;
+  listId?: string;
+  listTitle?: string;
+  id: string;
+  date?: string;
+}
+
+class SpoListItemDeclareRecord extends SpoCommand {
+
+  public allowUnknownOptions(): boolean | undefined {
+    return true;
+  }
+
+  public get name(): string {
+    return commands.LISTITEM_RECORD_DECLARE;
+  }
+
+  public get description(): string {
+    return "Declares the specified list item as a record";
+  }
+
+  public getTelemetryProperties(args: CommandArgs): any {
+    const telemetryProps: any = super.getTelemetryProperties(args);
+    telemetryProps.listId = typeof args.options.listId !== "undefined";
+    telemetryProps.listTitle = typeof args.options.listTitle !== "undefined";
+    telemetryProps.id = typeof args.options.id !== "undefined";
```

Comment:
The id option is required so this check would always return true, which is why there is no point in tracking its usage.

Reply:
Removed.